packages/js/product-editor/src/components/attribute-control/utils.ts

```diff
@@ -48,3 +53,36 @@ export function reorderSortableProductAttributePositions(
 		}
 	);
 }
+
+/**
+ * Checks if the given attribute has
+ * either terms (global attributes) or options (local attributes).
+ *
+ * @param {EnhancedProductAttribute} attribute - The attribute to check.
+ * @return {boolean} True if the attribute has terms or options, false otherwise.
+ */
+export const hasTermsOrOptions = (
+	attribute: EnhancedProductAttribute
+): boolean => {
+	return !! (
+		( attribute?.terms && attribute.terms.length > 0 ) ||
+		( attribute?.options && attribute.options.length > 0 )
```

Comment:
Can't this be simplified to `( attribute?.terms?.length > 0 ) || ( attribute?.options?.length > 0 )`

Reply:
probably, yes. Updating...

Reply:
actually `attribute?.terms?.length` can be `undefined`, impossible to compare with `> 0`.

we can do 
```
( attribute?.terms?.length || attribute?.options?.length )
```

Reply:
`undefined > 0` is false

Reply:
<img width="632" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/fbac86f8-2a07-4d75-b336-c09835301027">

The only way to address that is with something like...

```
( attribute?.name?.length ?? 0 ) > 0
```

I prefer the current code, tho

Reply:
Ah, TS... complicating valid JS. But, I know... it's just trying to look out for us! 😄 I'm good with the current code you went with... it is more understandable than the other workaround.

Reply:
Yes, it's better since we need a length different from zero. We don't need to check "mayor of".